code/game/objects/items/ego_weapons/teth.dm

```diff
@@ -312,3 +312,100 @@
+	special = "Attack nearby turfs to create traps. Remote mode can trigger traps from a distance. \
+	Automatic mode places traps that trigger when enemies walk over them. Use in hand to switch between modes."
+	icon_state = "lantern"
+	force = 8 //less than the baton, don't hit things with it
+	damtype = BLACK_DAMAGE
+	armortype = BLACK_DAMAGE
+
+	var/mode = LANTERN_MODE_REMOTE
+	var/traplimit = 6
+	var/list/traps = list()
+
+/obj/item/ego_weapon/lantern/attack_self(mob/user)
+	if(mode == LANTERN_MODE_REMOTE)
+		to_chat(user, "<span class='info'>You adjust any newly-placed traps to be set off by motion.</span>")
+		mode = LANTERN_MODE_AUTO
+	else
+		to_chat(user, "<span class='info'>You can now remotely trigger any placed traps.</span>")
+		mode = LANTERN_MODE_REMOTE
+
+/obj/item/ego_weapon/lantern/proc/CreateTrap(target, mob/user, proximity_flag)
+	var/turf/T = get_turf(target)
+	if((get_dist(user, T) > 20))
+		return
+	var/obj/effect/temp_visual/lanterntrap/R = locate(/obj/effect/temp_visual/lanterntrap) in T
+	if(R)
+		if(!proximity_flag && mode != LANTERN_MODE_REMOTE)
+			return
+		R.burst()
+		return
+	if(proximity_flag && (LAZYLEN(traps) < traplimit))
+		new /obj/effect/temp_visual/lanterntrap(T, user, src, mode)
+		user.changeNext_move(CLICK_CD_MELEE)
+
+/obj/item/ego_weapon/lantern/afterattack(atom/target, mob/living/user, proximity_flag, params)
+	if(check_allowed_items(target, 1))
+		CreateTrap(target, user, proximity_flag)
+	. = ..()
+
+/obj/effect/temp_visual/lanterntrap
+	name = "lantern trap"
+	icon_state = "shield1" //temp visual
+	layer = ABOVE_ALL_MOB_LAYER
+	duration = 30 SECONDS
+	var/resonance_damage = 35
+	var/damage_multiplier = 1
+	var/mob/creator
+	var/obj/item/ego_weapon/lantern/res
+	var/rupturing = FALSE //So it won't recurse
+
+/obj/effect/temp_visual/lanterntrap/Initialize(mapload, set_creator, set_resonator, mode)
+	if(mode == LANTERN_MODE_AUTO)
+		icon_state = "shield2" //temp visual
+		name = "lantern trap"
+		resonance_damage = 25
+		RegisterSignal(src, list(COMSIG_MOVABLE_CROSSED, COMSIG_ATOM_ENTERED), .proc/burst)
+	. = ..()
+	creator = set_creator
+	res = set_resonator
+	if(res)
+		res.traps += src
+	playsound(src,'sound/weapons/resonator_fire.ogg',50,TRUE)
+	deltimer(timerid)
+	timerid = addtimer(CALLBACK(src, .proc/burst), duration, TIMER_STOPPABLE)
+
+/obj/effect/temp_visual/lanterntrap/Destroy()
+	if(res)
+		res.traps -= src
+		res = null
+	creator = null
+	. = ..()
+
+/obj/effect/temp_visual/lanterntrap/proc/burst()
+	rupturing = TRUE
+	var/turf/T = get_turf(src)
+	new /obj/effect/temp_visual/resonance_crush(T) //temp visual
+	playsound(T,'sound/weapons/resonator_blast.ogg',50,TRUE)
+	for(var/mob/living/L in T)
+		if(creator.faction_check_mob(L))
+			continue
+		if(creator)
+			creator.visible_message("<span class='danger'>[creator] activates [src] on [L]!</span>","<span class='danger'>You activate [src] on [L]!</span>", null, COMBAT_MESSAGE_RANGE, L)
```

Comment:
This would make a visible message AT the creator? and it'd do it for every mob in the AoE? So you could have like 5 messages of it going off on like, a bunch of the fuckin worms at once. Or more if they're stacked. That's a lot of chat spam imo